### PR TITLE
First version of commons-ui 

### PR DESCRIPTION
**Pull request by mikatheo**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Prerequisites
```
Comment (geofjamg) [inline CONTRIBUTING.md:1]: It does not seem to be really a contributing file?
To remove?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@gridsuite/commons-ui",
+  "version": "1.0.0",
+  "description": "common-components React component",
```
Comment (geofjamg) [inline package.json:4]: We could write a better description maybe

File: tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import expect from 'expect'
```
Comment (geofjamg) [inline tests/index.test.js:1]: It does not seem to test anything?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "build": "nwb build-react-component --copy-files --no-demo",
+    "clean": "nwb clean-module && nwb clean-demo",
+    "prepublishOnly": "npm run build",
+    "start": "nwb serve-react-demo",
+    "test": "nwb test-react",
+    "test:coverage": "nwb test-react --coverage",
+    "test:watch": "nwb test-react --server"
+  },
+  "dependencies": {
+    "@material-ui/core": "latest",
```
Comment (geofjamg) [inline package.json:22]: We should not use latest dep as it produces instable builds

File: package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "@material-ui/icons": "^4.5.1",
+    "react-intl": "^3.11.0",
+    "react-request-fullscreen": "^1.1.2",
+    "prop-types": "latest"
```
Comment (geofjamg) [inline package.json:26]: We should not use latest dep as it produces instable builds

File: package.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  "peerDependencies": {
+    "react": "^16.x",
```
Comment (geofjamg) [inline package.json:29]: Why not a precise version?

File: package.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    "bufferutil": "^4.0.1",
+    "inferno": "^7.4.2",
+    "nwb": "^0.25.x",
+    "type-fest": "^0.13.1",
+    "utf-8-validate": "^5.0.2",
+    "webpack-plugin-serve": "^1.0.1"
+  },
+  "author": "",
+  "homepage": "",
+  "license": "MIT",
```
Comment (geofjamg) [inline package.json:43]: This is not the grdisuite licence

File: README.md [hunk 8]
```diff
@@ -1 +1,8 @@
-# common-components
\ No newline at end of file
+# commons-ui
+
+[![npm package][npm-badge]][npm]
```
Comment (geofjamg) [inline README.md:3]: ?

File: README.md [hunk 9]
```diff
@@ -1 +1,8 @@
...
+
+Library for sharing GridSuite apps commons components
+
+[npm-badge]: https://img.shields.io/npm/v/npm-package.png?style=flat-square
```
Comment (geofjamg) [inline README.md:7]: What is the purpose of a npm badge?

**Comment by geofjamg** [comment]
When I run the demo I have a lot of error like this:

```bash

 ERROR  in ./node_modules/react-transition-group/esm/ReplaceTransition.js
Module not found: Error: Can't resolve 'react-dom' in '/home/jamgotchiangeo/WebstormProjects/commons-ui/node_modules/react-transition-group/esm'
 @ ./node_modules/react-transition-group/esm/ReplaceTransition.js 5:0-33 92:56-64
 @ ./node_modules/react-transition-group/esm/index.js
 @ ./node_modules/@material-ui/core/esm/ButtonBase/TouchRipple.js
 @ ./node_modules/@material-ui/core/esm/ButtonBase/ButtonBase.js
 @ ./node_modules/@material-ui/core/esm/ButtonBase/index.js
 @ ./node_modules/@material-ui/core/esm/Button/Button.js
 @ ./node_modules/@material-ui/core/esm/Button/index.js
 @ ./src/components/TopBar/TopBar.js
 @ ./src/components/TopBar/index.js
 @ ./demo/src/index.js
 @ multi ./node_modules/@pmmmwh/react-refresh-webpack-plugin/src/runtime/ReactRefreshEntry.js (webpack)-dev-server/client?/ ./node_modules/@pmmmwh/react-refresh-webpack-plugin/src/runtime/ErrorOverlayEntry.js (webpack)/hot/only-dev-server.js ./demo/src/index.js
```

**Comment by geofjamg** [comment]
when running with npm start in IntelliJ, what is the URL to connect to the demo?
Isn't it supposed to automatically launch the browser with IntelliJ?

**Comment by ChamseddineBhd** [comment]
> When I run the demo I have a lot of error like this:
> 
> ```shell
>  ERROR  in ./node_modules/react-transition-group/esm/ReplaceTransition.js
> Module not found: Error: Can't resolve 'react-dom' in '/home/jamgotchiangeo/WebstormProjects/commons-ui/node_modules/react-transition-group/esm'
>  @ ./node_modules/react-transition-group/esm/ReplaceTransition.js 5:0-33 92:56-64
>  @ ./node_modules/react-transition-group/esm/index.js
>  @ ./node_modules/@material-ui/core/esm/ButtonBase/TouchRipple.js
>  @ ./node_modules/@material-ui/core/esm/ButtonBase/ButtonBase.js
>  @ ./node_modules/@material-ui/core/esm/ButtonBase/index.js
>  @ ./node_modules/@material-ui/core/esm/Button/Button.js
>  @ ./node_modules/@material-ui/core/esm/Button/index.js
>  @ ./src/components/TopBar/TopBar.js
>  @ ./src/components/TopBar/index.js
>  @ ./demo/src/index.js
>  @ multi ./node_modules/@pmmmwh/react-refresh-webpack-plugin/src/runtime/ReactRefreshEntry.js (webpack)-dev-server/client?/ ./node_modules/@pmmmwh/react-refresh-webpack-plugin/src/runtime/ErrorOverlayEntry.js (webpack)/hot/only-dev-server.js ./demo/src/index.js
> ```
Fixed, i have add     "react": "^16.13.1",
    "react-dom": "^16.13.1" as dev dependencies

**Pull request merged**